x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/upgrade_prebuilt_rules_table_context.tsx

```diff
@@ -61,15 +66,33 @@ export interface UpgradePrebuiltRulesTableState {
    * Rule rows selected in EUI InMemory Table
    */
   selectedRules: RuleUpgradeInfoForReview[];
+  /**
+   * Legacy ML Jobs that cause modal to pop up before performing rule ugprades
+   */
+  legacyJobsInstalled: MlSummaryJob[];
+  /**
+   * Upgrade method to execute when the user confirms the modal
+   */
+  modalConfirmationUpdateMethod: ModalConfirmationUpgradeMethod | null;
+  /**
+   * Is true when the modal is visible
+   * */
+  isUpgradeModalVisible: boolean;
```

Comment:
I propose we take this a step further and make those state props internal. The rules upgrade context should handle modal confirmation internally without revealing anything to the external environment. I might not have been clear enough about handling that state internally, so let's please sync up on this topic separately.